projects/components/src/search-box/search-box.component.ts

```diff
@@ -261,11 +264,22 @@ export class SearchBoxComponent implements OnInit, OnChanges {
     );
 
     this.subscriptionLifecycle.add(
-      this.valueChange.asObservable().subscribe(emittedValue => {
-        if (!isEmpty(emittedValue)) {
-          this.lastEmittedValues = [emittedValue, ...this.lastEmittedValues];
-        }
-      }),
+      this.debounceTime && this.debounceTime > 0 ?
+      this.valueChange
```

Comment:
Agree with the thought process here, but maybe we should just keep a large 1s or 2s debounce on valueChange. Should be fine. When the user is done with a meaningful search, they likely wait for a few seconds before searching again.

Reply:
I've set a constant debounce time. Which will be used by default if debounce time is not set. If debounce time is set, the minimum of the debounce time and the default debounce time is taken.